### PR TITLE
U36 weapon balance

### PR DESCRIPTION
**Pull request by ghost**
Weapon balance for CSP, Drak-25 and GK2

### CODE REVIEW

**Comment by ghost** [comment]
I forgot to change the arrayindex used as input to the Overclock function for ID and AV, so the version from the pull request has a bug.

File: src/weapons/driller/SludgePump.java [hunk 1]
```diff
@@ -35,21 +35,21 @@ public class SludgePump extends Weapon {
...
-	private double regularShotPuddleRadius;
-	private double chargedShotPuddleRadius;
-	private double corrosiveDoTDuration;
-	private double puddleLifetime;
+	private final double reloadTime;
+	private final double projectileVelocity;
+	private final double regularShotPuddleRadius;
+	private final double chargedShotPuddleRadius;
+	private final double corrosiveDoTDuration;
+	private final double puddleLifetime;
```
Comment (phg49389) [inline src/weapons/driller/SludgePump.java]: I would prefer it if these variables were not changed to be `final`.

File: src/weapons/scout/AssaultRifle.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ else if (selectedOverclock == 5) {
 	public StatsRow[] getStats() {
 		StatsRow[] toReturn = new StatsRow[12];
 		
-		boolean directDamageModified = selectedTier2 == 0 || selectedTier3 == 1 || selectedOverclock == 2 || selectedOverclock == 5 || selectedOverclock == 6;
+		boolean directDamageModified = selectedTier2 == 0 || selectedTier3 == 1 || selectedOverclock == 2 || selectedOverclock == 5;
```
Comment (phg49389) [inline src/weapons/scout/AssaultRifle.java:311]: Good catch on the StatsRow for the DirectDamage penalty no longer applying. In the PR, I don't see the corresponding change to indicate that the StatsRow for CarriedAmmo should change when Electrifying Reload is equipped.

**Review by phg49389 — CHANGES_REQUESTED** [review]
This is really good work! There was just one oversight in GK2 StatsRows that needs to get changed. I would prefer if the Sludge Pump variables weren't `final` but that's not a dealbreaker.

File: src/weapons/driller/SludgePump.java (hunk 1 above)
Comment (ghost) [reply src/weapons/driller/SludgePump.java]: I guess this was added when I was messing with IntelliJ. It was added by the IDE, I don't know exactly what "final" does.

**Pull request closed without merging**